pkg/inject/inject.go

```diff
@@ -656,15 +664,24 @@ func (conf *ResourceConfig) proxyUID() int64 {
 	return conf.proxyConfig.GetProxyUid()
 }
 
-func (conf *ResourceConfig) proxyProbe() *v1.Probe {
-	metricsPort := conf.proxyMetricsPort()
+func (conf *ResourceConfig) proxyReadinessProbe() *v1.Probe {
+	return &v1.Probe{
+		Handler: v1.Handler{
+			HTTPGet: &v1.HTTPGetAction{
+				Path: "/ready",
+				Port: intstr.IntOrString{IntVal: conf.proxyAdminPort()},
+			},
+		},
+		InitialDelaySeconds: 2,
+	}
+}
+
+func (conf *ResourceConfig) proxyLivenessProbe() *v1.Probe {
 	return &v1.Probe{
 		Handler: v1.Handler{
 			HTTPGet: &v1.HTTPGetAction{
 				Path: "/metrics",
-				Port: intstr.IntOrString{
-					IntVal: metricsPort,
-				},
+				Port: intstr.IntOrString{IntVal: conf.proxyAdminPort()},
```

Comment:
Should the `Path` be renamed to `/admin` too?

Reply:
No, we want the liveness probe to check the metrics endpoint still